blocks/api/parser.js

```diff
@@ -181,6 +181,26 @@ export function createBlockWithFallback( name, innerHTML, attributes ) {
 		// as invalid, or future serialization attempt results in an error
 		block.originalContent = innerHTML;
 
+		// If the block is invalid, try to find an older compatible version
```

Comment:
Maybe expand a bit on this comment, as it's important to understand why.

Quick one:
```
When a block is invalid, attempt to validate again using a supplied `deprecated` definition.
This allows blocks to modify their attribute and markup structure without invalidating
content written in previous formats.
```